ui/app/src/components/DateTimePicker.tsx

```diff
@@ -95,80 +128,124 @@ interface timezoneType {
+    controls = ['date', 'time', 'timezone'],
+    datePickerProps = {},
+    timePickerProps = {},
+    timeZonePickerProps = {}
+  } = props;
+  const [pickerState, setPickerState] = useState({
+    dateValid: true,
+    timeValid: true,
+    timezoneValid: true
+  });
+
+  // Set defaults on wrapped controls props
+  if (nou(timeZonePickerProps.timezone)) {
+    timeZonePickerProps['timezone'] = moment.tz.guess();
+  }
+  if (nou(datePickerProps.dateFormat)) {
+    datePickerProps['dateFormat'] = 'YYYY-MM-DD';
+  }
+  if (nou(datePickerProps.disablePast)) {
+    datePickerProps['disablePast'] = false;
+  }
+  if (nou(timePickerProps.timeFormat)) {
+    timePickerProps['timeFormat'] = 'HH:mm';
+  }
+
+  let dateMoment;
+  let timeMoment;
+  let timezoneObj = timezones.find(tz => (tz.timezoneName === unescape(timeZonePickerProps.timezone)));
+  dateMoment = getDateTimeMoment(date, timezoneObj);
+  timeMoment = getDateTimeMoment(date, timezoneObj);
+  const { formatMessage } = useIntl();
+
+  const handleDateChange = (name: string) => (newDate: Date | null) => {
+    let updatedDateTime = moment(newDate);
 
     switch (name) {
       case 'scheduledDate':
-        updatedDateTime.date(date.getDate());
-        updatedDateTime.month(date.getMonth());
-        updatedDateTime.year(date.getFullYear());
+        if (updatedDateTime._isValid) {
+          updatedDateTime.hours(timeMoment.toDate().getHours());
+          updatedDateTime.minutes(timeMoment.toDate().getMinutes());
+          updatedDateTime.seconds(timeMoment.toDate().getSeconds());
+          updatedDateTime.milliseconds(timeMoment.toDate().getSeconds());
+          datePickerProps?.onDateChange?.(updatedDateTime.format(datePickerProps.dateFormat));
+          setPickerState({ ...pickerState, dateValid: true });
+        } else {
+          setPickerState({ ...pickerState, dateValid: false });
+          onError?.();
+        }
         break;
       case 'scheduledTime':
-        updatedDateTime.hours(date.getHours());
-        updatedDateTime.minutes(date.getMinutes());
+        if (updatedDateTime._isValid) {
+          updatedDateTime.date(dateMoment.toDate().getDate());
+          updatedDateTime.month(dateMoment.toDate().getMonth());
+          updatedDateTime.year(dateMoment.toDate().getFullYear());
+          timePickerProps?.onTimeChange?.(updatedDateTime.format(timePickerProps.timeFormat));
+          setPickerState({ ...pickerState, timeValid: true });
+        } else {
+          setPickerState({ ...pickerState, timeValid: false });
+          onError?.();
+        }
         break;
     }
 
-    setSelectedDateTime(updatedDateTime);
-    onChange?.(updatedDateTime);
-
-    onChangeDate?.(updatedDateTime.format(dateFormat));
-    onChangeTime?.(updatedDateTime.format(timeFormat));
+    if (updatedDateTime._isValid) {
+      onChange?.(updatedDateTime);
+    }
   };
 
   const handleTimezoneChange = () => (event: React.ChangeEvent<{}>, timezoneObj: any) => {
```

Comment:
No point this being a creator instead of a straight function. 

Below, where this is used, the second param would only ever comes if the autocomplete sends these two a params.